### PR TITLE
Fix #465 for isapprox with complex arrays

### LINKED ISSUE #465: `isapprox` for complex unitful arrays fails
MWE for v1.8.0:

```julia
using Unitful: mm
z = (1+im) * ones(2,3) * mm
isapprox(z, 2z)

ERROR: MethodError: no method matching isless(::ComplexF64, ::ComplexF64)
...
```

It works fine for a complex scalar `z = (1 + im) * mm`
This is perhaps slightly related to #310 but should be an easier case.

### PR DESCRIPTION
**Pull request by JeffFessler**
Fix #465
The substantive change was `real{T1}` on line 300.

The rest of it is just white space to make it easier for me to sort out why it worked for scalars and not arrays.
The scalar case on line 282 has `real(T)` in it, so now it is matching between scalar and array.

This seems like a bug fix so I put in v1.9.1, but if you prefer v1.10.0 because it now it is new support for complex arrays that's fine too.  Up to you...

### CODE REVIEW

File: src/quantities.jl [hunk 1]
```diff
@@ -275,18 +275,31 @@ for (i,j) in zip((:<, :isless), (:_lt, :_isless))
...
-isapprox(x::AbstractQuantity{T,D,U}, y::AbstractQuantity{T,D,U}; atol=zero(Quantity{real(T),D,U}), kwargs...) where {T,D,U} =
-    isapprox(x.val, y.val; atol=uconvert(unit(y), atol).val, kwargs...)
+
+function isapprox(
+    x::AbstractQuantity{T,D,U},
+    y::AbstractQuantity{T,D,U};
+    atol = zero(Quantity{real(T),D,U}),
+    kwargs...,
+) where {T,D,U}
+    return isapprox(x.val, y.val; atol=uconvert(unit(y), atol).val, kwargs...)
```
Comment (sostock) [inline src/quantities.jl:285]:
```suggestion
    return isapprox(x.val, y.val; atol=ustrip(unit(y), atol), kwargs...)
```

File: test/runtests.jl [hunk 2]
```diff
@@ -741,6 +741,10 @@ end
         @test isapprox(1.0u"m", 1.1u"m"; rtol=0.2)
         @test !isapprox(1.0u"m", 1.1u"m"; rtol=0.05)
 
+        # Issue 465:
+        z = (1+im) * ones(2,3) * 1m
+        @test !isapprox(z, 2z)
```
Comment (giordano) [inline test/runtests.jl:746]: Maybe also add a positive test like `isapprox(z, z)`?

File: src/quantities.jl (hunk 1 above)
Comment (JeffFessler) [reply src/quantities.jl:285]: I'll commit this suggestion, but I should mention that the `uconvert` version was already in there.

File: test/runtests.jl (hunk 2 above)
Comment (JeffFessler) [reply test/runtests.jl:746]: Done

**Comment by JeffFessler** [comment]
bump - I think I have addressed all the suggestions last summer.  Then I forgot about it, but I just hit this same issue again and "rediscovered" this PR.
Hopefully the only remaining question is whether to call it v0.11.0 or v0.10.1 (I chose the latter).

File: test/runtests.jl [hunk 3]
```diff
@@ -760,6 +760,11 @@ end
         @test isapprox(1.0u"m", 1.1u"m"; rtol=0.2)
         @test !isapprox(1.0u"m", 1.1u"m"; rtol=0.05)
 
+        # Issue 465:
+        z = (1+im) * ones(2,3) * 1m
```
Comment (sostock) [inline test/runtests.jl:764]:
```suggestion
        z = fill((1+im)m, 2, 3)
```
I think this would be clearer than the multiplication chain.

**Review by sostock — APPROVED** [review]
LGTM. We should resolve some more issues with `isapprox`(#310, ~~support for the `nans` keyword~~), but that can happen in a future PR.

File: test/runtests.jl (hunk 3 above)
Comment (JeffFessler) [reply test/runtests.jl]: agreed!

**Pull request merged**